api/src/main/java/com/epam/pipeline/manager/cluster/autoscale/ScaleDownHandler.java

```diff
@@ -448,4 +453,15 @@ private Optional<String> getTimestampTag(final String name) {
                 .filter(StringUtils::isNotEmpty)
                 .map(suffix -> name + suffix);
     }
+
+    private Duration findNodeUnavailableGraceDuration(final Optional<PipelineRun> pipelineRun,
+                                                      final Duration defaultDuration) {
+        return pipelineRun
```

Comment:
What will happen if `NODE_UNAVAILABLE_GRACE_PERIOD_MINUTES` is not correct? (f.e. empty or string instead of number)
let's double check that we will not fail here, and if we have something inappropriate (like `str` or int < 0), let's return default value here